libraries/default.rb

```diff
@@ -735,25 +735,32 @@ def enable_newrelic?
       node[:newrelic]
     end
 
-    def configure_newrelic(current_deploy_root, node_name)
-      if enable_newrelic?
-        log_dir = node.fetch(:matterhorn_log_directory, '/var/log/matterhorn')
-
-        newrelic_att = node.fetch(:newrelic, {})
-        newrelic_key = newrelic_att[:key]
-        environment_name = node[:opsworks][:stack][:name]
-        template %Q|#{current_deploy_root}/etc/newrelic.yml| do
-          source 'newrelic.yml.erb'
-          owner 'matterhorn'
-          group 'matterhorn'
-          variables({
-            newrelic_key: newrelic_key,
-            node_name: node_name,
-            environment_name: environment_name,
-            log_dir: log_dir
-          })
-        end
-      end
+    def enable_newrelic_layer?(layer_name)
+       nr_layers = node.fetch(:newrelic,{}) #  must be a hash
+       nr_layers.key?(layer_name)
+    end
+
+
+    def configure_newrelic(current_deploy_root, node_name, layer_name)
+      if enable_newrelic_layer?(layer_name)
+            log_dir = node.fetch(:matterhorn_log_directory, '/var/log/matterhorn')
+            newrelic_layers = node.fetch(:newrelic,{})
+            newrelic_layer = newrelic_layers.fetch(layer_name,{})
+            newrelic_key = newrelic_layer[:key]
+
+            environment_name = node[:opsworks][:stack][:name]
```

Comment:
Is it ok for this value to contain whitespace? if not, you should use the `stack_shortname` helper.